components/raftstore/src/store/fsm/peer.rs

```diff
@@ -5491,15 +5508,21 @@ where
 
     fn on_schedule_half_split_region(
         &mut self,
+        start_key: Option<Vec<u8>>,
+        end_key: Option<Vec<u8>>,
         region_epoch: &metapb::RegionEpoch,
         policy: CheckPolicy,
         source: &str,
         _cb: Callback<EK::Snapshot>,
     ) {
+        let start_key_to_log = start_key.clone().unwrap_or_default();
+        let end_key_to_log = end_key.clone().unwrap_or_default();
         info!(
-            "on half split";
+            "on half split key range";
             "region_id" => self.fsm.region_id(),
             "peer_id" => self.fsm.peer_id(),
+            "start_key" => log_wrappers::Value::key(&start_key_to_log),
```

Comment:
if not specified the `start key` or `end key`,  will it be empty here? but actually, it should be the region's key range.

Reply:
The exact key is calculated in [check_split_and_bucket](https://github.com/tikv/tikv/blob/2accd27c2a3882d3d45b6958565efb6de4d6e33c/components/raftstore/src/store/worker/split_check.rs#L322-L323). To reuse as much of the old code as possible, we have to keep the key as `Option` pass-through until we get there.